tests/sqlite3/test-sqlite3.cpp

```diff
@@ -237,6 +237,44 @@ TEST_CASE("SQLite vector long long", "[sqlite][vector][longlong]")
     CHECK(v2[4] == 1000000000000LL);
 }
 
+struct type_inference_table_creator : table_creator_base
+{
+    type_inference_table_creator(soci::session & sql)
+        : table_creator_base(sql)
+    {
+        sql << "create table soci_test(cvc varchar (10), cdec decimal (20), "
+               "cll bigint, cull unsigned bigint, clls big int, culls unsigned big int)";
+    }
+};
+
+// test for correct type inference form sqlite column type
+TEST_CASE("SQLite type inference", "[sqlite][sequence]")
+{
+    soci::session sql(backEnd, connectString);
+
+    type_inference_table_creator tableCreator(sql);
+
+    std::string cvc = "john";
+    double cdec = 12345.0;  // integers stores preciesly in IEEE 754
+    long long cll = 1000000000003LL;
+    unsigned long long cull = 1000000000004ULL;
+
+    sql << "insert into soci_test(cvc, cdec, cll, cull, clls, culls) values(:cvc, :cdec, :cll, :cull, :clls, :culls)",
+        use(cvc), use(cdec), use(cll), use(cull), use(cll), use(cull);
+
+    {
+        rowset<row> rs = (sql.prepare << "select * from soci_test");
+        rowset<row>::const_iterator it = rs.begin();
+        row const& r1 = (*it);
+        CHECK(r1.get<std::string>(0) == cvc);
+        CHECK(r1.get<double>(1) == cdec);
```

Comment:
You can use CATCH `Approx` for comparing doubles to avoid the warning which resulted in CI build failures.